crates/solidity/inputs/language/src/dsl.rs

```diff
@@ -0,0 +1,915 @@
+
+    scanner DoubleQuotedAsciiStringLiteral = ('"' ((EscapeSequence | ((' ' .. '!') | ('#' .. '[') | (']' .. '~'))) *) '"') ;
+
+    scanner DoubleQuotedHexStringLiteral = ("hex" '"' (HexStringContents ?) '"') ;
+
+    scanner DoubleQuotedUnicodeStringLiteral = { enabled from "0.7.0" ("unicode\"" ((EscapeSequence | (! "\n\r\"\\")) *) '"') } ;
+
+    scanner ElseKeyword = "else" ;
+
+    scanner EmitKeyword = { enabled from "0.4.21" "emit" } ;
+
+    scanner EndOfLine = (('\r' ?) '\n') ;
+
+    scanner EnumKeyword = "enum" ;
+
+    scanner Equal = '=' ;
+
+    scanner EqualEqual = "==" ;
+
+    scanner EqualGreaterThan = "=>" ;
+
+    scanner ErrorKeyword = { enabled from "0.8.4" "error" } ;
+
+    scanner EscapeSequence = ('\\' (AsciiEscape | HexByteEscape | UnicodeEscape)) ;
+
+    scanner EtherKeyword = "ether" ;
+
+    scanner EventKeyword = "event" ;
+
+    scanner ExperimentalKeyword = "experimental" ;
+
+    scanner ExternalKeyword = "external" ;
+
+    scanner FallbackKeyword = "fallback" ;
+
+    scanner FalseKeyword = "false" ;
+
+    scanner FinneyKeyword = { disabled from "0.7.0" "finney" } ;
+
+    scanner FixedBytesType = ("bytes" FixedBytesTypeSize) ;
+
+    scanner FixedBytesTypeSize = (
+        "1" | "10" | "11" | "12" | "13" | "14" | "15" | "16" | "17" | "18" | "19" | "2" | "20" | "21" | "22" | "23" |
+        "24" | "25" | "26" | "27" | "28" | "29" | "3" | "30" | "31" | "32" | "4" | "5" | "6" | "7" | "8" | "9"
+    ) ;
+
+    scanner FixedTypeSize = ((('0' .. '9') +) 'x' (('0' .. '9') +)) ;
+
+    scanner ForKeyword = "for" ;
+
+    scanner FromKeyword = "from" ;
+
+    scanner FunctionKeyword = "function" ;
+
+    scanner GlobalKeyword = "global" ;
+
+    scanner GreaterThan = '>' ;
+
+    scanner GreaterThanEqual = ">=" ;
+
+    scanner GreaterThanGreaterThan = ">>" ;
+
+    scanner GreaterThanGreaterThanEqual = ">>=" ;
+
+    scanner GreaterThanGreaterThanGreaterThan = ">>>" ;
+
+    scanner GreaterThanGreaterThanGreaterThanEqual = ">>>=" ;
+
+    scanner GweiKeyword = { enabled from "0.6.11" "gwei" } ;
+
+    scanner HexByteEscape = ('x' (HexCharacter) (HexCharacter)) ;
+
+    scanner HexCharacter = (('0' .. '9') | ('A' .. 'F') | ('a' .. 'f')) ;
+
+    scanner HexLiteral = (("0x" | {disabled from "0.5.0" "0X"}) ((HexCharacter) +) (('_' ((HexCharacter) +)) *)) ;
+
+    scanner HexStringContents = ((HexCharacter) (HexCharacter) ((('_' ?) (HexCharacter) (HexCharacter)) *)) ;
+
+    scanner HexStringLiteral = (SingleQuotedHexStringLiteral | DoubleQuotedHexStringLiteral) ;
+
+    scanner HoursKeyword = "hours" ;
+
+    scanner Identifier = RawIdentifier ;
+
+    scanner IdentifierPart = (IdentifierStart | ('0' .. '9')) ;
+
+    scanner IdentifierStart = ('$' | ('A' .. 'Z') | '_' | ('a' .. 'z')) ;
+
+    scanner IfKeyword = "if" ;
+
+    scanner ImmutableKeyword = { enabled from "0.6.5" "immutable" } ;
+
+    scanner ImportKeyword = "import" ;
+
+    scanner IndexedKeyword = "indexed" ;
+
+    scanner IntegerTypeSize = (
```

Comment:
Does this sorting impact the scanner? I wonder if we should maintain the original order for readability?

Reply:
Not at all. Sure.